prod.Dockerfile

```diff
@@ -0,0 +1,54 @@
+FROM python:3.12-slim
+
+# The uv installer requires curl (and certificates) to download the release archive
+RUN apt-get update; \
+    apt-get install -y --no-install-recommends ca-certificates curl vim; \
+    apt-get autoremove --purge -y; \
+    apt-get clean -y; \
+    rm -rf /var/lib/apt/lists/* /var/cache/apt/archives/*
+
+# Download the latest installer
+ADD https://astral.sh/uv/install.sh /uv-installer.sh
+
+# Run the uv installer then remove it
+RUN sh /uv-installer.sh && rm /uv-installer.sh
+
+# Ensure the installed binary is on the `PATH`
+ENV PATH="/root/.local/bin/:$PATH"
+
+# Install the project into `/app`
+WORKDIR /app
+
+# Enable bytecode compilation
+ENV UV_COMPILE_BYTECODE=1
+
+# Copy from the cache instead of linking since it's a mounted volume
+ENV UV_LINK_MODE=copy
+
+# Install the project's dependencies using the lockfile and settings
+RUN --mount=type=cache,target=/root/.cache/uv \
+    --mount=type=bind,source=uv.lock,target=uv.lock \
+    --mount=type=bind,source=pyproject.toml,target=pyproject.toml \
+    uv sync --frozen --no-install-project --no-dev && uv pip install pip
+
+
+# Then, add the rest of the project source code and install it
+# Installing separately from its dependencies allows optimal layer caching
+COPY . /app
+RUN --mount=type=cache,target=/root/.cache/uv \
+    uv sync --frozen --no-dev
+
+# Place executables in the environment at the front of the path
+ENV PATH="/app/.venv/bin:$PATH"
+
+# Running gunicorn with Uvicorn workers
+CMD [ \
+    "gunicorn", \
```

Comment:
I remember you mentioned this before but I forgot -- what's the reason for using `gunicorn` _and_ `uvicorn` here?

It looks odd as `gunicorn` is a WSGI server, so we wouldn't be getting any of the benefits of `asyncio` unless I'm missing something?

Reply:
it's one of the suggested way to run uvicorn in production. See [https://www.uvicorn.org/deployment/#gunicorn](https://www.uvicorn.org/deployment/#gunicorn).